zerver/lib/mention.py

```diff
@@ -218,6 +218,12 @@ def possible_user_group_mentions(content: str) -> set[str]:
     return {m.group("match") for m in USER_GROUP_MENTIONS_RE.finditer(content)}
 
 
+def non_silent_user_group_mentions(content: str) -> set[str]:
+    return {
+        m.group("match") for m in USER_GROUP_MENTIONS_RE.finditer(content) if not m.group("silent")
+    }
```

Comment:
Let's avoid doing the regexp match twice; just make `possible_user_group_mentions` return a dataclass with two attributes.

Reply:
Agree